_posts/2018-06-18-configuration.md

```diff
@@ -0,0 +1,160 @@
+
+__Runtime Configuration__
+
+If you've ever used a system environment variable to configure some part of your application, then you're familiar with runtime configuration.
+As the name suggests, this is the configuration for an application at the time it is run.
+These are the values we can expect to change as we deploy our build artifacts to different systems.
+
+__Application Configuration__
+
+Application configuration is something different, and the difference, while subtle, can be a pitfall in certain situations.
+The difference shines when we consider that our code, and it's configuration, is compiled into a build artifact we can distribute.
+
+We can say with certainty that no matter where our application is run there are certain things we want to remain constant.
+No matter where we deploy our application, we want know we want to use the same `Logger` configuration.
+If we're relying on dependency injection for testing, then we know for certain we don't want to use those dependencies in the final deliverable.
+They configure the function of our code.
+
+### How it's been done
+
+The frustration for many people has to do with the usage of `Application.get_env/2` and `System.get_env/1`.
+We'll jump right in by looking at a configuration that's common to many Elixir projects:
+
+```elixir
+use Mix.Config
+
+config :example_app, Data.Repo,
+  adapter: Ecto.Adapters.MySQL,
+  username: System.get_env("EXAMPLE_APP_USERNAME"),
+  password: System.get_env("EXAMPLE_APP_PASSWORD"),
+  hostname: System.get_env("EXAMPLE_APP_HOSTNAME"),
+  database: System.get_env("EXAMPLE_APP_DATABASE"),
+  pool_size: 10
+```
+
+Simple and harmless enough, right?
+Wrong!
+
+Our application's configuration, defined in `config.exs` and friends, is compiled when we generate build artifacts, like those produced by Distillery.
+That means those `System.get_env/1` functions need to be resolved at compile time.
+See the problem?
+Our application's code is coupled to our system configuration.
+
+What if we want to generate the build artifact locally and run it elsewhere?
+What if there's an emergency and the value of `EXAMPLE_APP_HOSTNAME` has been updated?
+With this configuration our application needs to be precompile for changes to take effect.
+
+Let's illustrate the concept using colors to differeniate changes:
+
+![elixir-config-recompile](https://user-images.githubusercontent.com/73386/41503026-d8a66ef4-7185-11e8-95fa-37598f6a56ff.png)
+
+Here we see that our runtimes values are different, which required we recompile our code.
+This results in a new build artifact and updated configuration for Runtime B.
+We've managed to couple the runtime configuration and code together.
+To see changes in our environment reflected in our code recompilation is unavoidable.
+For those using Releases this combobulation of configuration types often times requires additional libraries to fill the gaps.
```

Comment:
Did you mean to capitalize "Releases"?

Will "combobulation" make it difficult to translate this article into other languages, since the word is slang? If so, "combination" may be a better fit.

Reply:
This is a very valid point, I will change it to "combination"